### ISSUE TITLE
BasicKV::PrefixLookup:  I notice the founction may return one KV pair. it is designing for that or a bug?

### ISSUE THREAD

**Issue by SYaoJun** [zz-jason/leanstore#133]
## expection
if I insert five pairs: 
"hello": 1, 
 "hey": 2, 
 "hi": 3, 
 "holy":4,  
"haha": 5

the prefixlookup("he") should return 2 kv pairs:
"hello": 1, 
 "hey": 2, 
but the current code maybe return one kv pair, or return all the ascscan KV pairs.  

## code
![image](https://github.com/user-attachments/assets/6496791a-3a81-40cf-876c-66f31bc85287)

**Comment by zz-jason** [comment]
Right, BasicKv hasn't been fully tested. Would you like to fix this issue?

**Comment by SYaoJun** [comment]
yes, I can follow the BasicKV::ScanAsc() function, which use a pessimistic method. Actually, I want to fix it based on optimistic method. But I don't know how to handle the situation when a leaf node need move to their neighbor leaf

**Comment by zz-jason** [comment]
When latch conflict happens (the page being scanned is modified by others at the same time), an optimistic scan needs to:
- retry from the beginning (or retry scanning from the upper bound keys of the last unchanged page)
- clear all the previously scanned and buffered key values

The current API sends each scanned key value to a custom callback, which makes the retry hard to implement. The easiest way to correctly implement ScanDesc or ScanAsc is to use a pessimistic shared iterator.
